lib/dynflow/config.rb

```diff
@@ -32,15 +32,51 @@ def validate
         @config.validate(self)
       end
 
+      def queues
+        @queues ||= @config.queues.finalized_config(self)
+      end
+
       def method_missing(name)
         return @cache[name] if @cache.key?(name)
         value = @config.send(name)
         value = value.call(@world, self) if value.is_a? Proc
-        @config.send("validate_#{ name }!", value)
+        validation_method = "validate_#{ name }!"
+        @config.send(validation_method, value) if @config.respond_to?(validation_method)
         @cache[name] = value
       end
     end
 
+    class QueuesConfig
+      attr_reader :queues
+
+      def initialize
+        @queues = {:default => {}}
+      end
+
+      # Add a new queue to the configuration
+      #
+      # @param [Hash] queue_options
+      # @option queue_options :pool_size The amount of workers available for the queue.
+      #   By default, it uses global pool_size config option.
+      def add(name, queue_options = {})
+        Utils.validate_keys!(queue_options, :pool_size)
+        name = name.to_sym
+        raise ArgumentError, "Queue #{name} is already defined" if @queues.key?(name)
+        @queues[name] = queue_options
+      end
+
+      def finalized_config(config_for_world)
+        @queues.values.each do |queue_options|
```

Comment:
Here we should probably check if the world is an executor world. For client worlds it doesn't make much sense to set queue sizes for them as they can't actually do anything. I know this has no real effect on the functionality, but it might be confusing when shown in the ui.

Reply:
I would not check if the config has executor here, because the executor configuration is using this method, so I would get into endless recursion. However, I updated the code to not pass the metadata about queues if the world doesn't have an executor defined.